msal/src/main/java/com/microsoft/identity/client/statemachine/states/ResetPasswordStates.kt

```diff
@@ -0,0 +1,374 @@
+                PublicApiId.NATIVE_AUTH_RESET_PASSWORD_RESEND_CODE
+            )
+
+            val rawCommandResult = CommandDispatcher.submitSilentReturningFuture(command).get()
+
+            return@withContext when (val result = rawCommandResult.checkAndWrapCommandResultType<ResetPasswordResendCodeCommandResult>()) {
+                is ResetPasswordCommandResult.CodeRequired -> {
+                    ResetPasswordResendCodeResult.Success(
+                        nextState = ResetPasswordCodeRequiredState(
+                            flowToken = result.passwordResetToken,
+                            config = config
+                        ),
+                        codeLength = result.codeLength,
+                        sentTo = result.challengeTargetLabel,
+                        channel = result.challengeChannel
+                    )
+                }
+
+                is INativeAuthCommandResult.Redirect -> {
+                    ResetPasswordResult.BrowserRequired(
+                        error = BrowserRequiredError(
+                            correlationId = result.correlationId
+                        )
+                    )
+                }
+
+                is INativeAuthCommandResult.UnknownError -> {
+                    Logger.warn(
+                        TAG,
+                        "Unexpected result: $result"
+                    )
+                    ResetPasswordResult.UnexpectedError(
+                        error = GeneralError(
+                            errorMessage = result.errorDescription,
+                            error = result.error,
+                            correlationId = result.correlationId,
+                            details = result.details,
+                            exception = result.exception
+                        )
+                    )
+                }
+            }
+        }
+    }
+}
+
+/**
+ * Native Auth uses a state machine to denote state and transitions for a user.
+ * ResetPasswordPasswordRequiredState class represents a state where the user has to provide a password to progress
+ * in the reset password flow.
+ * @property flowToken: Flow token to be passed in the next request
+ * @property config Configuration used by Native Auth
+ */
+class ResetPasswordPasswordRequiredState internal constructor(
+    override val flowToken: String,
+    private val config: NativeAuthPublicClientApplicationConfiguration
+) : BaseState(flowToken), State, Serializable {
+    private val TAG: String = ResetPasswordPasswordRequiredState::class.java.simpleName
+
+    interface SubmitPasswordCallback : Callback<ResetPasswordSubmitPasswordResult>
+
+    /**
+     * Submits a new password to the server; callback variant.
+     *
+     * @param password The password to submit.
+     * @param callback [com.microsoft.identity.client.statemachine.states.ResetPasswordPasswordRequiredState.SubmitPasswordCallback] to receive the result on.
```

Comment:
Is this going to link to the SubmitPasswordCallback file from within the Javadoc? Generally the `@link` is used to create these links

Reply:
Yes, this will work. Kotlin allows linking with square brackets. https://kotlinlang.org/docs/kotlin-doc.html#inline-markup